addons/gamification/tests/test_challenge.py

```diff
@@ -59,6 +61,67 @@ def test_10_reach_challenge(self):
         badge_ids = self.env['gamification.badge.user'].search([('badge_id', '=', badge_id), ('user_id', '=', demo.id)])
         self.assertEqual(len(badge_ids), 1, "Demo user has not received the badge")
 
+    @mute_logger('odoo.models.unlink')
+    def test_20_update_all_goals_filter(self):
+        challenge = self.env.ref('gamification.challenge_base_discover')
+
+        # Enroll two internal and two portal users in the challenge
+        self.env.ref('base.demo_user0').copy()
+        all_test_users = self.env['res.users'].search([], order="share desc", limit=4)
+        portal_login_before_update, portal_login_after_update = all_test_users.filtered(lambda u: u.share)
+        internal_login_before_update, internal_login_after_update = all_test_users.filtered(lambda u: not u.share)
```

Comment:
as it depends on the search, it is a bit weird ... isn't it possible to have a custom setup data ? A bit more data creation but at least we will have a clean base for the future, when we will rewrite everthing (ah bon ?) !